packages/dashboard-v2/src/pages/user/confirm.js

```diff
@@ -0,0 +1,68 @@
+import { useEffect, useState } from "react";
+import { navigate } from "gatsby";
+
+import { AllUsersAuthLayout } from "../../layouts/AuthLayout";
+
+import HighlightedLink from "../../components/HighlightedLink";
+import accountsService from "../../services/accountsService";
+
+const State = {
+  Pure: "PURE",
+  Success: "SUCCESS",
+  Failure: "FAILURE",
+};
+
+const EmailConfirmationPage = ({ location }) => {
+  const query = new URLSearchParams(location.search);
+  const token = query.get("token");
+
+  const [state, setState] = useState(State.Pure);
+
+  useEffect(() => {
+    let timer;
+
+    async function confirm(token) {
+      try {
+        await accountsService.get("user/confirm", { searchParams: { token } });
+
+        timer = setTimeout(() => {
+          navigate("/");
+        }, 3000);
+        setState(State.Success);
+      } catch {
+        setState(State.Failure);
+      }
+    }
+
+    if (token) {
+      confirm(token);
+    }
+
+    return () => clearTimeout(timer);
```

Comment:
This can run with timer being undefined if it rerenders with different token before `accountsService.get` resolves - probably totally unlikely but still a good practice to fix that I think.

I would make the hook return function async, assign `const confirming = await confirm()` and `await confirmation`  in the return function to make sure that `timer` is set before we clear it.

Reply:
Yeah, extremely unlikely given the `token` comes from `location` :)

But better be on the safe side I guess.
I fixed it by adding an `AbortController`. Now if the token changes we'll cancel the previous request. Cancelling a request raises an error, so the `timer` will never be set (which means we don't need to clear it).

👉  be82050b